test/normalCommand.test.ts

```diff
@@ -0,0 +1,74 @@
+import { newTest } from './testSimplifier';
+import { cleanUpWorkspace, setupWorkspace } from './testUtils';
+
+suite('Execute normal command', () => {
+  setup(async () => {
+    await setupWorkspace();
+  });
+
+  teardown(cleanUpWorkspace);
+
+  newTest({
+    title: 'One liner',
+    start: ['foo =| bar = 1'],
+    keysPressed: ':normal f=i!=\n',
+    end: ['foo = bar !|== 1'],
+  });
+
+  newTest({
+    title: 'One liner with selection',
+    start: ['foo =| bar = 1'],
+    keysPressed: 'V:normal f=i!=\n',
+    end: ['foo !|== bar = 1'],
+  });
+
+  newTest({
+    title: 'Multiple liner with selection',
+    start: ['foo =| bar = 1', 'foo = bar = 2'],
+    keysPressed: 'Vj:normal f=i!=\n',
+    end: ['foo !== bar = 1', 'foo !|== bar = 2'],
+  });
+
+  newTest({
+    title: 'Multiple liner with line range',
+    start: ['foo =| bar = 1', 'foo = bar = 2', 'foo = bar = 3'],
+    keysPressed: ':2,3normal f=i!=\n',
+    end: ['foo = bar = 1', 'foo !== bar = 2', 'foo !|== bar = 3'],
+  });
+
+  newTest({
+    title: 'One liner with dot',
+    start: ['foo =| bar = 1', 'foo = bar = 2'],
+    keysPressed: 'f=i!=<Esc>j^f=:normal .\n',
+    end: ['foo = bar !== 1', 'foo !|== bar = 2'],
+  });
+
+  // TODO: remove comment out after fixing dot command bug
+  // newTest({
+  //   title: 'One liner with multiple dot',
+  //   start: ['foo =| bar = 1', 'foo = bar = 2'],
+  //   keysPressed: 'f=i!=<Esc>j^f=:normal 2.\n',
+  //   end: ['foo = bar !== 1', 'foo !=!|== bar = 2'],
```

Comment:
I commented out this test because dot command behavior of insert action is different from original vim one.
This is a bug of dot command...
https://github.com/VSCodeVim/Vim/issues/8938